src/main/java/com/mongodb/hibernate/internal/translate/SelectMqlTranslator.java

```diff
@@ -57,6 +62,12 @@ public JdbcOperationQuerySelect translate(
                 renderMongoAstNode(aggregateCommand),
                 getParameterBinders(),
                 jdbcValuesMappingProducer,
-                getAffectedTableNames());
+                getAffectedTableNames(),
+                0,
+                Integer.MAX_VALUE,
+                Collections.emptyMap(),
+                JdbcLockStrategy.NONE,
+                getFirstRowParameter(),
+                getMaxRowsParameter());
```

Comment:
The above stuff has to be passed from translator (I borrowed from `AbstractSqlAstTranslator`); otherwise the dynamic `QueryOptions` Limit value won't be accessible. Why? I am still figuring out, but we can prove that by integratin testing case.